src/AnnotationThread.js

```diff
@@ -113,94 +130,148 @@ class AnnotationThread extends EventEmitter {
         this.state = STATES.inactive;
     }
 
-    /**
-     * Whether or not thread dialog is visible
-     *
-     * @return {void}
-     */
-    isDialogVisible() {
-        return !!(this.dialog && this.dialog.element && !this.dialog.element.classList.contains(CLASS_HIDDEN));
+    position = () => {
+        /* eslint-enable no-unused-vars */
+        throw new Error('Implement me!');
+    };
+
+    getPopoverParent() {
+        return util.shouldDisplayMobileUI(this.container)
+            ? this.container
+            : util.getPageEl(this.annotatedElement, this.location.page);
     }
 
     /**
      * Shows the appropriate annotation dialog for this thread.
      *
+     * @param {Event} event - Mouse event
      * @return {void}
      */
-    showDialog() {
-        // Prevents the annotations dialog from being set up on every call
-        if (!this.dialog.element) {
-            this.dialog.setup(this.annotations, this.element);
+    renderAnnotationPopover(event = null) {
+        if (event) {
+            event.stopPropagation();
+            event.preventDefault();
         }
 
-        this.dialog.show();
+        const isPending = this.state === STATES.pending;
+        this.onCommentClick = this.onCommentClick.bind(this);
+        this.save = this.save.bind(this);
+        this.updateTemporaryAnnotation = this.updateTemporaryAnnotation.bind(this);
+        this.delete = this.delete.bind(this);
```

Comment:
Can these be bound elsewhere/constructor? It's possible that every time this is called on an instance, you can end up in scope hell